enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,515 @@
+Allowing specific IP addresses is achieved through the use of `NetworkPolicy`
+resources, as seen in previous examples. These policies include the allowed IP
+addresses and ports in the spec, meaning they reject everything else that is
+not included explicitly.
+
+Due to MicroShift's networking setup, there is one more change required for
+the `NetworkPolicy` to filter out based on the source IP.
+
+As it can be seen in the iptables rules, traffic towards the router ports is
+subject to DNAT to the router service IP and port. All services use
+`externalTrafficPolicy: Cluster`, which will SNAT traffic's source IP to match
+that of the node. This gets further NAT, as described here. By the time the
+`NetworkPolicy` is evaluated the source IP has been replaced with
+`100.64.0.0/16`, which will not produce matches against the configured IP in
+the policy, and traffic will get rejected.
+
+To overcome this, all LoadBalancer services in MicroShift that are subject to
+get NetworkPolicy resources to control external traffic must set
+`externalTrafficPolicy: Local`. This setting will not SNAT packets and the
+source IP is preserved until NetworkPolicy checks run.
+
+This setting, however, comes at the expense of an even load when having more
+than one node, as traffic will not be sent to another node on the same port.
+This is not an issue in single node, which is MicroShift's use case and design
+principle.
+
+This setting is also the default in OCP when deploying in cloud, having a load
+balancer before the cluster. Manifests for both OCP and MicroShift are
+compatible in this regard.
+
+## Open Questions
+N/A
+
+## Test Plan
+All of the changes listed here will be included in the current e2e scenario
+testing harness in MicroShift.
+
+## Graduation Criteria
+Targeting GA for MicroShift 4.16 release.
+
+### Dev Preview -> Tech Preview
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+
+### Tech Preview -> GA
+- More testing (upgrade, downgrade)
+- Sufficient time for feedback
+- Available by default
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+### Removing a deprecated feature
+N/A
+
+## Upgrade / Downgrade Strategy
+In the previous implementation, using host ports, the router would listen on
+any interface (except loopback). This included some of the internal IP
+addresses from ovnk, such as `169.254.169.2`, which is the link local IP
+address that ovnk uses in `br-ex` for internal handling of flows.
+
+The approach described in this enhancement does not expose the router in such
+IP, as it introduces conflicts between iptables rules and openflow.
+
+## Version Skew Strategy
+N/A
+
+## Operational Aspects of API Extensions
+
+### Failure Modes
+If the configured entries in `ingress.expose.ipAddresses` and
+`ingress.expose.interfaces` do not exist in the node, MicroShift should fail
+to start.
```

Comment:
Can I set `status` to "Disabled" and still specify `ports` or `expose`?

What happens in case of a DNS lookup failure?

What happens if I specify a port that is already in use?

Reply:
I'm curious whether MicroShift will refuse to start if I put an invalid IP address, NIC, or whatever _and_ set `status` to "Disabled".  I suppose it isn't necessary to go into detail about that case here though.

Reply:
It would be good to be clear about the precedence. If the status is disabled, I would expect everything else to be ignored.

Reply:
Done.